include/nvexec/stream/scan.cuh

```diff
@@ -0,0 +1,172 @@
+
+          void* d_temp_storage{};
+          std::size_t temp_storage_size{};
+
+          auto first = begin(range);
+          auto last = end(range);
+
+          std::size_t num_items = std::distance(first, last);
+
+          if (status = STDEXEC_DBG_ERR(cub::DeviceScan::ExclusiveScan(
+                d_temp_storage,
+                temp_storage_size,
+                first,
+                d_out,
+                self.fun_,
+                self.init_,
+                num_items,
+                stream));
+              status != cudaSuccess) {
+            self.op_state_.propagate_completion_signal(stdexec::set_error, std::move(status));
+            return;
+          }
+
+          if (status = STDEXEC_DBG_ERR( //
+                cudaMallocAsync(&d_temp_storage, temp_storage_size, stream));
+              status != cudaSuccess) {
+            self.op_state_.propagate_completion_signal(stdexec::set_error, std::move(status));
+            return;
+          }
+
+          if (status = STDEXEC_DBG_ERR(cub::DeviceScan::ExclusiveScan(
+                d_temp_storage,
+                temp_storage_size,
+                first,
+                d_out,
+                self.fun_,
+                self.init_,
+                num_items,
+                stream));
+              status != cudaSuccess) {
+            self.op_state_.propagate_completion_signal(stdexec::set_error, std::move(status));
+            return;
+          }
+
+          status = STDEXEC_DBG_ERR(cudaFreeAsync(d_temp_storage, stream));
+          self.op_state_.defer_temp_storage_destruction(d_out);
+          if (status == cudaSuccess) {
+            int* host_out = new int[10];
+            cudaMemcpy(host_out, d_out, 10 * sizeof(int), cudaMemcpyDeviceToHost);
+            for (size_t i = 0; i < 10; ++i) {
+              std::cout << host_out[i] << std::endl;
+            }
+            self.op_state_.propagate_completion_signal(stdexec::set_value, d_out);
+          } else {
+            self.op_state_.propagate_completion_signal(stdexec::set_error, std::move(status));
+          }
+        }
+      };
+
+      template <class SenderId, class InitT, class Fun>
+      struct sender_t
+        : public __algo_range_init_fun::
+            sender_t<SenderId, InitT, Fun, sender_t<SenderId, InitT, Fun>> {
+        template <class Receiver>
+        using receiver_t =
+          stdexec::__t<scan_::receiver_t<SenderId, stdexec::__id<Receiver>, InitT, Fun>>;
+
+        template <class Range>
+        using _set_value_t = completion_signatures<set_value_t(
+          ::std::add_lvalue_reference_t<
+            typename __algo_range_init_fun::binary_invoke_result_t<Range, InitT, Fun>>)>;
+
+        // template <class Range>
+        // using _set_value_t = completion_signatures<set_value_t(
+        //   std::vector<typename __algo_range_init_fun::binary_invoke_result_t<Range, InitT, Fun>>)>;
+        
```

Comment:
Not sure how to get the completion signatures right. Hoping to get some guidance